### ISSUE TITLE
Marten SQL queries that dont start with select cause error

### ISSUE THREAD

**Issue by DanCymonz** [JasperFx/marten#2753]
I discussed this on the discord as well. I found a work around but had another idea as well.

There was a change to marten in v6 on how it detects if its running a sql query or something else. This causes an application error when you have sql queries that dont start with select. Common table expressions is a good example as they start with "with". I found a work around by wrapping the with query in another select but maybe a slightly better solution would be for the check to change to start with "select" or "with". It still feels a bit clunky but it is slightly better. 

![CleanShot 2023-10-15 at 09 59 28](https://github.com/JasperFx/marten/assets/106283559/649a2077-acfe-419d-810f-59db39cfef1f)

https://github.com/JasperFx/marten/commit/08b86f49d1e02db393461d29bce070ee9fac7813

![CleanShot 2023-10-15 at 09 54 49](https://github.com/JasperFx/marten/assets/106283559/45ad3862-9e89-456c-9913-016d2cdc062a)

The check is used here which causes the error
![CleanShot 2023-10-15 at 10 01 37](https://github.com/JasperFx/marten/assets/106283559/08b7d53b-25a9-4259-8f2b-03cd00285981)

**Comment by oskardudycz** [comment]
@DanCymonz, could you provide more reproduce steps explaining your use case? Configuration, sample query, etc.? Best would be the failing test to our codebase.

**Comment by danyo1399** [comment]
yeah i was thinking about maybe fixing it myself since the change im proposing is so trivial. leave it with me for now. The hardest part will prolly be getting the tests running locally lol

**Comment by DanCymonz** [comment]
yeah i was thinking about maybe fixing it myself since the change im proposing is so trivial. leave it with me for now. The hardest part will prolly be getting the tests running locally.

**Comment by oskardudycz** [comment]
> yeah i was thinking about maybe fixing it myself since the change im proposing is so trivial. leave it with me for now. The hardest part will prolly be getting the tests running locally.

To run tests, it should be enough to just run:

````bash
docker-compose up
````
and run tests either from IDE or via `dotnet test` nothing magical besides that is needed.
